wearipedia/devices/whoop/whoop_gen.py

```diff
@@ -1,163 +1,573 @@
+    syn_collection["start"] = start_list
+    syn_collection["end"] = end_list
+    syn_collection["timezone_offset"] = "-08:00"
+    syn_collection["nap"] = [False] * num_days
+    syn_collection["score_state"] = "SCORED"
+    syn_collection["score"] = [{} for _ in range(num_days)]
+
+    for i in range(num_days):
+        # Initialize scores for the day
+        total_sleep_time = int(np.random.normal(2.88e7, 3.6e6))
+        light_sleep_proportion = np.random.normal(0.55, 0.05)
+        slow_wave_sleep_proportion = np.random.normal(0.22, 0.05)
+        rem_sleep_proportion = np.random.normal(0.23, 0.05)
+        awake_proportion = np.random.normal(0.07, 0.01)
+
+        # Normalize proportions to ensure they sum to 1
+        total_proportion = (
+            light_sleep_proportion
+            + slow_wave_sleep_proportion
+            + rem_sleep_proportion
+            + awake_proportion
         )
-        cur_timestamp += timedelta(seconds=60.563)
+        light_sleep_proportion /= total_proportion
+        slow_wave_sleep_proportion /= total_proportion
+        rem_sleep_proportion /= total_proportion
+        awake_proportion /= total_proportion
+
+        scores = {
+            "stage_summary": {
+                "total_in_bed_time_milli": total_sleep_time,
+                "total_awake_time_milli": int(total_sleep_time * awake_proportion),
+                "total_no_data_time_milli": 0,
+                "total_light_sleep_time_milli": int(
+                    total_sleep_time * light_sleep_proportion
+                ),
+                "total_slow_wave_sleep_time_milli": int(
+                    total_sleep_time * slow_wave_sleep_proportion
+                ),
+                "total_rem_sleep_time_milli": int(
+                    total_sleep_time * rem_sleep_proportion
+                ),
+                "sleep_cycle_count": int(np.random.normal(3, 0.5)),
+                "disturbance_count": int(np.random.normal(5, 2)),
+            },
+            "sleep_needed": {
+                "baseline_milli": int(np.random.normal(25000000, 500000)),
+                "need_from_sleep_debt_milli": int(np.random.uniform(0, 500000)),
+                "need_from_recent_strain_milli": int(np.random.uniform(0, 500000)),
+                "need_from_recent_nap_milli": int(np.random.uniform(-50000, 0)),
+            },
+            "respiratory_rate": np.round(np.random.normal(15, 2), 2),
+            "sleep_performance_percentage": int(spp(total_sleep_time)),
+            "sleep_consistency_percentage": int(np.random.normal(90, 5)),
+            "sleep_efficiency_percentage": np.round(np.random.normal(90, 5), 2),
+        }
+
+        # Set the scores for the day
+        syn_collection.at[i, "score"] = scores
+
+    return syn_collection
+
+
+def spp(x):
+    a = (90 - 78) / (2.88e7 - 1.80e7)
+    b = 78 - a * 1.80e7
+    y = a * x + b + int(np.random.uniform(-3.5, 3.5))
+    return min(100, y)
+
+
+# Define the probability distribution
```

Comment:
It might be a good idea to jus define the prob_distribution and ranges variables within get_strain, as it they are not accessed outside of the scope of that function

Reply:
Done!